main.go

```diff
@@ -41,9 +43,30 @@ func main() {
 		panic(fmt.Sprintf("Cannot run with a thread number greater than %d", tempThreadLimit))
 	}
 
+	yamlParam := "yaml"
+	yamlValue := dummyYAML
+	yamlUsage := "the path to the YAML configuration file which will supersede all other flags"
+	yaml := flag.String(yamlParam, yamlValue, yamlUsage)
+
 	// parsing the flags has to be done after setting up all the flags
 	flag.Parse()
 
+	var (
```

Comment:
No need to declare variables like this within code. First, it looks very ugly :( Second, you can use the type inference of Go to write: 
```Go
cgf, err := config_parser.ParseConfig(*yaml).
```

Reply:
The reason for declaring variables here separately is because we need `cfg` to be in scope to line 76 where new master process is created.

Reply:
See comment for line 55.